sanic/handlers/error.py

```diff
@@ -23,16 +20,21 @@ class ErrorHandler:
     by the developers to perform a wide range of tasks from recording the error
     stats to reporting them to an external service that can be used for
     realtime alerting system.
-
     """
 
+    DEFAULT_HANDLERS = {
+        (SanicIsADirectoryError, None): DirectoryHandler.default_handler
+    }
+
     def __init__(
         self,
         base: Type[BaseRenderer] = TextRenderer,
     ):
         self.cached_handlers: Dict[
             Tuple[Type[BaseException], Optional[str]], Optional[RouteHandler]
-        ] = {}
+        ] = {
+            **self.DEFAULT_HANDLERS  # type: ignore
+        }
```

Comment:
I'm thinking rather than this being the pattern, we should register in the app instance so that:
1. custom error handlers still have this feature 
2. we can make the directory handler a passed arg like error handler is for overriding